src/wallet/rpcdump.cpp

```diff
@@ -1482,3 +1482,263 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
 
     return response;
 }
+
+static UniValue ProcessDescriptorImport(CWallet * const pwallet, const UniValue& data, const int64_t timestamp) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
+{
+    UniValue warnings(UniValue::VARR);
+    UniValue result(UniValue::VOBJ);
+
+    try {
+        if (!data.exists("desc")) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Descriptor not found.");
+        }
+
+        const std::string& descriptor = data["desc"].get_str();
+        const bool active = data.exists("active") ? data["active"].get_bool() : true;
+        const bool internal = data.exists("internal") ? data["internal"].get_bool() : false;
+        const std::string& label = data.exists("label") ? data["label"].get_str() : "";
+        const std::string& output_param = data.exists("address_type") ? data["address_type"].get_str() : "";
+
+        OutputType output_type = pwallet->m_default_address_type;
+        if (!ParseOutputType(output_param, output_type)) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Unknown address type '%s'", output_param));
+        }
+
+        // Active descriptor must have a range
+        if (active && !data.exists("range")) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Active descriptor must have a range");
+        }
+
+        // Internal addresses should not have a label
+        if (internal && data.exists("label")) {
```

Comment:
I'm not sure that it also makes sense to have a label for active descriptor either. If something is going to be active, then the newly generated addresses can get their own labels too. Also users may not be expecting that their new addresses will have a label that they set at the beginning.

We could either outright disallow `active` and `label`, or, combined with a `next_index` parameter, set the label for everything from the range start to (but not including) the next index.

Reply:
I'm still unclear on how labels fit in with descriptors in general...

1/
> We could either outright disallow `active` and `label`
If we disallow `active` and `label` together, what happens if the user sets the `label` first, _then_ change the descriptor to be `active`? Would they then have to "undo" the `label`? Seems messy.

2/ 
> Also users may not be expecting that their new addresses will have a label that they set at the beginning.
It's also unintuitive IMHO for a label to only apply up to a certain index. That means users who want to use labels would have to remember to set them every once a while when the index runs out. Doesn't seem user-friendly?

More importantly, I actually think all this `label` stuff should be moved entirely into the UI/application layer like QT. Seems weird that a UI concept gets implemented at the Core layer at all.

Reply:
> If we disallow `active` and `label` together, what happens if the user sets the `label` first, _then_ change the descriptor to be `active`? Would they then have to "undo" the `label`? Seems messy.

That shouldn't be possible since that would require importing the same descriptor twice which we don't allow.

> It's also unintuitive IMHO for a label to only apply up to a certain index. That means users who want to use labels would have to remember to set them every once a while when the index runs out. Doesn't seem user-friendly?

Currently every new address does not have a label by default (or rather it has the empty string as a label). IMO users should expect that they need to set labels on newly generated addresses, which, in this case, are addresses from `next_index` and beyond. Since everything from range start up to but not including `next_index` are considered to be already used, it would make sense for those to have the label that they set here.

Perhaps this is something that we should bring up at the next IRC meeting.

> More importantly, I actually think all this `label` stuff should be moved entirely into the UI/application layer like QT. Seems weird that a UI concept gets implemented at the Core layer at all.

Well it has to be stored somewhere, and it makes more sense to put it in the wallet.

Reply:
> That shouldn't be possible since that would require importing the same descriptor twice which we don't allow.

Right. Silly me.

> Since everything from range start up to but not including next_index are considered to be already used, it would make sense for those to have the label that they set here.

What if they had multiple labels for the used indices?

> Well it has to be stored somewhere, and it makes more sense to put it in the wallet.

Yeah that I agree. What I'm saying is maybe that logic of the wallet could be in the UI-facing layer, and not the core wallet layer? Labels seem like superficial cosmetic stuff applying on addresses, which is unlike other fields we (historically) track.

Reply:
I think maybe it would be better if we just disallowed ranged descriptors and labels. If someone wants their imported addresses to have labels, they can set them after the import using `setlabel`.